pkg/resource/deploy/builtins.go

```diff
@@ -230,3 +239,23 @@ func (p *builtinProvider) readStackResourceOutputs(inputs resource.PropertyMap)
 		"outputs": resource.NewObjectProperty(outputs),
 	}, nil
 }
+
+func (p *builtinProvider) readStackResource(inputs resource.PropertyMap) (resource.PropertyMap, error) {
+	urn, ok := inputs["urn"]
+	if !ok {
+		return nil, fmt.Errorf("readStackResource missing required 'urn' argument")
+	}
+	if !urn.IsString() {
+		return nil, fmt.Errorf("readStackResource 'urn' argument expected string got %s", urn.TypeString())
+	}
+
+	state := p.plan.current[resource.URN(urn.StringValue())]
+	if state == nil {
+		return nil, fmt.Errorf("resource '%s' not yet registered in current stack state", urn.StringValue())
+	}
+
+	return resource.PropertyMap{
+		"urn":     resource.NewStringProperty(string(state.URN)),
```

Comment:
1. technically seems redundant right?  the caller passed in the urn, so passing back out seems unnecessary.
2. should we actually assert these are the same.  seems like it would be super bad if they weren't

Reply:
> technically seems redundant right? the caller passed in the urn, so passing back out seems unnecessary.

Yes - it is.

> should we actually assert these are the same. seems like it would be super bad if they weren't

Yes - we should.